astropy/units/quantity_helper/erfa.py

```diff
@@ -98,6 +99,24 @@ def helper_gc2gd(f, nounit, unit1):
         )
 
 
+def helper_gc2gde(f, unit_r, unit_flat, unit_xyz):
+    from astropy.units.si import m, radian
+
+    if unit_r is None:
```

Comment:
I don't think this is necessary - just let `get_converter` handle this!

I realize the error messages may not be ideal, but I think it may be better in that case to try to improve `get_converter` -- but in a different PR!